lib/wire_client/base/common_message_entity.rb

```diff
@@ -0,0 +1,43 @@
+module WireClient
+  class CommonMessageEntity
+    include ActiveModel::Validations
+    extend Converter
+
+    attr_accessor :name,
+                  :iban,
+                  :bic,
+                  :account_number,
+                  :wire_routing_number,
+                  :clear_system_code,
+                  :postal_code,
+                  :address_line,
+                  :city,
+                  :country_subdivision,
+                  :country,
+                  :currency
+
+    convert :name, to: :text
+    validates_length_of :currency, is: 3
+    validates_length_of :name, within: 1..70
+    validates_with CurrencyValidator,
+                   CountryValidator,
+                   CountrySubdivisionValidator,
+                   BICValidator,
+                   IBANValidator,
+                   message: "%{value} is invalid"
+
+    def country_subdivision_abbr
+      if @country == 'US' && !@country_subdivision.match(/\A[A-Z]{2,2}\z/)
+        return US_STATES[@country_subdivision]
+      end
+      @country_subdivision
+    end
+
+    def country_subdivision_name
+      if @country == 'US' && @country_subdivision.match(/\A[A-Z]{2,2}\z/)
```

Comment:
Can we create a separate method for these two conditions?
We would be able to use these methods in both `country_subdivision_abbr` and `country_subdivision_name`